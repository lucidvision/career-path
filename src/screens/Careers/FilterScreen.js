import React, { Component } from 'react'
import { Picker, StyleSheet, Slider, Text, View } from 'react-native'
import { Header, IconButton } from 'components'
import { Feather } from '@expo/vector-icons'
import { jobs } from 'data/jobs'
import _ from 'lodash'

class FilterScreen extends Component {
  static navigationOptions = {
    title: 'Filter'
  }
  state = {
    minSalary: 10000,
    maxSalary: 200000,
    education: 'Bachelors'
  }
  handleFilterPressed = () => {
    const { minSalary, maxSalary, education } = this.state
    if (minSalary || maxSalary || education) {
      const results = _.filter(jobs, job => {
        const { salary, educationRequired } = job
        let keep = true
        if (minSalary && +salary < +minSalary) {
          keep = false
        }
        if (maxSalary && +salary > +maxSalary) {
          keep = false
        }
        if (education && !(education === educationRequired)) {
          keep = false
        }
        return keep
      })
      this.props.navigation.navigate('CareersList', {name: 'Results', jobs: results})
    }
  }
  render () {
    return (
      <View style={styles.container}>
        <Header style={styles.verticleSpacing} text='Minimum Salary' />
        <Slider
          maximumValue={100000}
          minimumValue={10000}
          onValueChange={minSalary => this.setState({minSalary})}
          step={1000}
          style={styles.slider}
          value={this.state.minSalary} />
        <Text style={styles.verticleSpacing}>{this.state.minSalary}</Text>
        <Header style={styles.verticleSpacing} text='Maximum Salary' />
        <Slider
          maximumValue={200000}
          minimumValue={10000}
          onValueChange={maxSalary => this.setState({maxSalary})}
          step={1000}
          style={styles.slider}
          value={this.state.maxSalary} />
        <Text style={styles.verticleSpacing}>{this.state.maxSalary}</Text>
        <Header text='Education' />
        <Picker
          style={styles.picker}
          selectedValue={this.state.education}
          onValueChange={education => this.setState({education})}>
          <Picker.Item label='High School' value='High School' />
          <Picker.Item label='Bachelors' value='Bachelors' />
          <Picker.Item label='Masters' value='Masters' />
          <Picker.Item label='PhD' value='PhD' />
        </Picker>
        <IconButton onPress={this.handleFilterPressed} title='Filter'>
          <Feather name='filter' size={25} />
        </IconButton>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    padding: 10
  },
  picker: {
    width: '100%'
  },
  slider: {
    width: '100%'
  },
  verticleSpacing: {
    marginBottom: 20
  }
})

export default FilterScreen
