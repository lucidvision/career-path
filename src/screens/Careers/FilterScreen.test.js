import React from 'react'
import { shallow } from 'enzyme'
import FilterScreen from './FilterScreen'

const results = {
  jobs: [
    {
      educationRequired: 'Bachelors',
      name: 'Nurse',
      salary: 50000,
      score: 1
    },
    {
      educationRequired: 'Bachelors',
      name: 'Therapist',
      salary: 50000,
      score: 0
    },
    {
      educationRequired: 'Bachelors',
      name: 'Agent',
      salary: 50000,
      score: 0
    },
    {
      educationRequired: 'Bachelors',
      name: 'Hotel Manager',
      salary: 80000,
      score: 1
    }
  ],
  name: 'Results'
}

const createFilterScreen = (override) => {
  const props = {...override}

  return shallow(<FilterScreen {...props} />)
}

describe('FilterScreen', () => {
  const component = createFilterScreen()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  describe('Min Salary', () => {
    it('has a header for min salary', () => {
      expect(component.find('Header').at(0).props().text).toBe('Minimum Salary')
    })

    it('sets state for min salary using slider', () => {
      component.find('Slider').at(0).props().onValueChange(20000)
      expect(component.state().minSalary).toBe(20000)
    })

    it('sets min salary text', () => {
      component.update()
      expect(component.find('Text').at(0).props().children).toBe(20000)
    })
  })

  describe('Max Salary', () => {
    it('has a header for min salary', () => {
      expect(component.find('Header').at(1).props().text).toBe('Maximum Salary')
    })

    it('sets state for min salary using slider', () => {
      component.find('Slider').at(1).props().onValueChange(100000)
      expect(component.state().maxSalary).toBe(100000)
    })

    it('sets min salary text', () => {
      component.update()
      expect(component.find('Text').at(1).props().children).toBe(100000)
    })
  })

  describe('Education', () => {
    it('has a header for education', () => {
      expect(component.find('Header').at(2).props().text).toBe('Education')
    })

    it('sets state for education using picker', () => {
      component.find('Picker').props().onValueChange('Masters')
      expect(component.state().education).toBe('Masters')
    })
  })

  describe('handleFilterPressed', () => {
    it('navigates to careers list with results', () => {
      const navMock = jest.fn()
      const component = createFilterScreen({
        navigation: {
          navigate: navMock
        }
      })
      component.find('IconButton').props().onPress()
      expect(navMock).toHaveBeenCalledWith('CareersList', results)
    })
  })
})
