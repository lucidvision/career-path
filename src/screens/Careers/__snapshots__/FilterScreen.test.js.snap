// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Header
    style={
      Object {
        "marginBottom": 20,
      }
    }
    text="Minimum Salary"
  />
  <Slider
    disabled={false}
    maximumValue={100000}
    minimumValue={10000}
    onValueChange={[Function]}
    step={1000}
    style={
      Object {
        "width": "100%",
      }
    }
    value={10000}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    10000
  </Text>
  <Header
    style={
      Object {
        "marginBottom": 20,
      }
    }
    text="Maximum Salary"
  />
  <Slider
    disabled={false}
    maximumValue={200000}
    minimumValue={10000}
    onValueChange={[Function]}
    step={1000}
    style={
      Object {
        "width": "100%",
      }
    }
    value={200000}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    200000
  </Text>
  <Header
    text="Education"
  />
  <Picker
    mode="dialog"
    onValueChange={[Function]}
    selectedValue="Bachelors"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <PickerItem
      label="High School"
      value="High School"
    />
    <PickerItem
      label="Bachelors"
      value="Bachelors"
    />
    <PickerItem
      label="Masters"
      value="Masters"
    />
    <PickerItem
      label="PhD"
      value="PhD"
    />
  </Picker>
  <IconButton
    onPress={[Function]}
    title="Filter"
  >
    <Icon
      allowFontScaling={false}
      name="filter"
      size={25}
    />
  </IconButton>
</View>
`;
